src/main/java/io/kestra/plugin/hightouch/connections/Sync.java

```diff
@@ -0,0 +1,213 @@
+
+        // Trigger sync run
+        HttpResponse<Run> triggerResponse = this.request(
+                "POST",
+                UriTemplate
+                        .of("/api/v1/syncs/{syncId}/trigger")
+                        .expand(Map.of(
+                                "syncId", runContext.render(this.syncId.toString())
+                        )),
+                String.format(
+                        "{\"fullResync\": %s}",
+                        runContext.render(this.fullResynchronization.toString())
+                ),
+                Run.class
+        );
+
+        Run jobInfoRead = triggerResponse.getBody().orElseThrow(() -> new IllegalStateException("Missing body on trigger"));
+
+        logger.info("Job status {} with response: {}", triggerResponse.getStatus(), jobInfoRead);
+        Long runId = jobInfoRead.getId();
+
+        if (!this.wait) {
+            return Output.builder()
+                .runId(runId)
+                .build();
+        }
+
+        // wait for end
+        RunDetails finalJobStatus = Await.until(
+            throwSupplier(() -> {
+                        HttpResponse<RunDetailsResponse> detailsResponse = this.request(
+                                "GET",
+                                UriTemplate
+                                        .of("/api/v1/syncs/{syncId}/runs?runId={runId}")
+                                        .expand(Map.of(
+                                                "syncId", runContext.render(this.syncId.toString()),
+                                                "runId", runId
+                                        )),
+                                "{}",
+                                RunDetailsResponse.class
+                        );
+
+                RunDetailsResponse runDetailsResponse = detailsResponse.getBody().orElseThrow(() -> new IllegalStateException("Missing body on trigger"));
+
+                // Check we correctly get one run
+                if (runDetailsResponse.getData().isEmpty()) {
+                    logger.warn("Could not find the triggered runId {}", runId);
+                    throw new Exception("Failed : could not find the triggered runId : " + runId);
+                }
+
+                // Illegal state where we have more than 1 item
+                if (runDetailsResponse.getData().size() > 1) {
+                    logger.warn("Found several entries for runId {}", runId);
+                    throw new Exception("Failed: found several runs with runId : " + runId);
+                }
+
+                RunDetails runDetails = runDetailsResponse.getData().get(0);
+                sendLog(logger, syncDetails, runDetails);
+
+                // ended
+                if (ENDED_STATUS.contains(runDetails.getStatus())) {
+                    return runDetails;
+                }
+                return null;
+            }),
+            Duration.ofSeconds(2),
```

Comment:
```suggestion
            Duration.ofSeconds(1),
```
Can you also set it as a constant field of the class so it's easier to change it if needed